packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -619,6 +629,65 @@ foreach(plugin ${FLUTTER_PLUGIN_LIST})
 endforeach(plugin)
 ''';
 
+const String _dartPluginRegisterWith = r'''
+      try {
+        {{dartClass}}.registerWith();
+      } catch (err) {
+        print(
+          '`{{pluginName}}` threw an error: $err. '
+          'The app may not function as expected until you remove this plugin from pubspec.yaml'
+        );
+      }
+''';
+
+// TODO(egarciad): Evaluate merging the web and desktop plugin registry templates.
+// https://github.com/flutter/flutter/issues/80406
+const String _dartPluginRegistryForDesktopTemplate = '''
+//
+// Generated file. Do not edit.
+//
+
+// @dart = {{dartLanguageVersion}}
+
+import '{{mainEntrypoint}}' as entrypoint;
+import 'dart:io'; // flutter_ignore: dart_io_import.
+{{#linux}}
+import 'package:{{pluginName}}/{{pluginName}}.dart';
+{{/linux}}
+{{#macos}}
+import 'package:{{pluginName}}/{{pluginName}}.dart';
+{{/macos}}
+{{#windows}}
+import 'package:{{pluginName}}/{{pluginName}}.dart';
+{{/windows}}
+
+@pragma('vm:entry-point')
+class _PluginRegistrant {
+
+  @pragma('vm:entry-point')
+  static void register() {
+    if (Platform.isLinux) {
+      {{#linux}}
+$_dartPluginRegisterWith
+      {{/linux}}
+    } else if (Platform.isMacOS) {
+      {{#macos}}
+$_dartPluginRegisterWith
+      {{/macos}}
+    } else if (Platform.isWindows) {
+      {{#windows}}
+$_dartPluginRegisterWith
+      {{/windows}}
+    }
+  }
+
+}
+
+void main() {
```

Comment:
For desktop applications this will drop the `List<String> args`. You could update this to check the arity of `main` and then conditionally pass args through. See `flutter run -a, --dart-entrypoint-args` for how to use these from the CLI.

Reply:
this is a tricky one. One could have a `void main()` within a string literal.  I would need to think about this one.

Reply:
I would do any checks of the string contents of the file. You could do a type check of the main method and then decide how to invoke it. For example, see: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/isolated/resident_web_runner.dart#L709

Reply:
* wouldn't

Reply:
ah - that's great workaround. Done.